src/js/line.js

```diff
@@ -0,0 +1,517 @@
+            // Whether or not to add a point to each datapoint forming the line
+            addPoints: false,
+            pointRadius: 2,
+            // In milliseconds
+            transitionLength: 2000
+        },
+        styles: {
+            svg: "floe-ca-lineChart-chart",
+            chartLinePoint: "floe-ca-lineChart-chartLinePoint"
+        },
+        selectors: {
+            title: ".floec-ca-lineChart-title",
+            description: ".floec-ca-lineChart-description",
+            svg: ".floec-ca-lineChart-line",
+            yAxis: ".floec-ca-lineChart-y-axis",
+            xAxis: ".floec-ca-lineChart-x-axis",
+            chartLine: ".floec-ca-lineChart-chartLine",
+            chartLinePointGroup: ".floec-ca-lineChart-chartLinePointGroup",
+            chartLinePoint: ".floec-ca-lineChart-chartLinePoint",
+            chartLineArea: ".floec-ca-lineChart-chartLineArea"
+        },
+        events: {
+            onChartCreated: null,  // Fire when the line is created. Ready to register D3 DOM event listeners,
+            onChartRedrawn: null // Fire when the chart is redrawn.
+        },
+        listeners: {
+            "onCreate.create": {
+                funcName: "floe.chartAuthoring.lineChart.chart.create",
+                args: ["{that}"]
+            }
+        },
+        modelListeners: {
+            dataSet: {
+                funcName: "{that}.draw",
+                excludeSource: "init"
+            }
+        },
+        modelRelay: {
+            source: "{that}.model.dataSet",
+            target: "{that}.model.wrappedDataSet",
+            singleTransform: {
+                type: "fluid.transforms.free",
+                args: ["{that}.model.dataSet"],
+                func: "floe.chartAuthoring.lineChart.chart.wrapSingleDataSet"
+            }
+        },
+        invokers: {
+            draw: {
+                funcName: "floe.chartAuthoring.lineChart.chart.draw",
+                args: ["{that}"]
+            }
+        }
+    });
+
+    // Test if a dataset is multi
+    floe.chartAuthoring.lineChart.chart.isMultiDataSet = function (dataSet) {
+        return dataSet[0].id !== undefined && dataSet[0].data !== undefined;
+    };
+
+    // Wrap a non-multi dataset (a simple array without ID keys) so we can
+    // process it with the same function used for multi datasets, and
+    // create a default id for it for object constancy
+    floe.chartAuthoring.lineChart.chart.wrapSingleDataSet = function (dataSet) {
+        if (!floe.chartAuthoring.lineChart.chart.isMultiDataSet(dataSet)) {
+            dataSet = [{
+                id: "dataSet",
+                data: dataSet
+            }];
+        }
+        return dataSet;
+    };
+
+    floe.chartAuthoring.lineChart.chart.draw = function (that) {
+
+        var shouldAddArea = that.options.lineOptions.addArea,
+            shouldAddPoints = that.options.lineOptions.addPoints;
+
+        floe.chartAuthoring.lineChart.chart.addYAxis(that);
```

Comment:
These should be expressed as invokers so that the component can be easily customised